src/helpers/lang.ts

```diff
@@ -5,52 +5,63 @@ import ptBr from '../assets/locales/pt-BR/lang.json'
 import { LocalStorage } from '~/helpers/local-storage'
 import { isString } from '~/helpers/data'
 import { setHtmlAttr } from '~/helpers/dom'
+import { Setup } from '~/types/Setup'
 
 export enum Lang {
     EN_US = 'en-US',
     PT_BR = 'pt-BR',
 }
 
-const LANG_KEY = import.meta.env.VITE_LANG_KEY
-const DEFAULT_LANG = Lang.EN_US
-const setLang = (lang: Lang) => LocalStorage.set<string>(LANG_KEY, lang)
+export class LangHelper implements Setup<Lang> {
+    readonly DEFAULT_VALUE = Lang.EN_US
+    private readonly LANG_KEY = import.meta.env.VITE_LANG_KEY
 
-const validLang = (propName: string): Lang | null => {
-    if (!isString(propName)) propName = ''
+    init() {
+        i18n.use(initReactI18next).init({
+            resources: {
+                [Lang.EN_US]: { translation: enUs },
+                [Lang.PT_BR]: { translation: ptBr },
+            },
+            lng: this.get(),
+            fallbackLng: Lang.EN_US,
+            interpolation: {
+                escapeValue: false,
+            },
+        })
+    }
 
-    let lang: Lang | null = null
-    const hasPropNameLength = propName.length >= 2
-    const initialPropName = propName.slice(0, 2)
+    get(): Lang {
+        const lang =
+            this.getValidLangByPropName(LocalStorage.get<string>(this.LANG_KEY)) ||
+            this.getValidLangByPropName(navigator.language) ||
+            this.DEFAULT_VALUE
 
-    if (!hasPropNameLength) return lang
-    if (Lang.EN_US.startsWith(initialPropName)) lang = Lang.EN_US
-    if (Lang.PT_BR.startsWith(initialPropName)) lang = Lang.PT_BR
+        this.set(lang)
```

Comment:
I think it's strange a get function who else set something

Reply:
I'll fix, thanks